### PR TITLE
Color: Add guidelines for color usage

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Add new page for Color Usage, update Color to Color palette, add guidance on when to use which colors

#### Why?

Provide guidance and documentation around our color tokens. This new guidance also warranted a color update for secondary and tertiary background colors

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-3528)
- [Figma](https://www.figma.com/file/Lacd4Q9LkpX4KRqVOKLQnA/Color-usage?node-id=2%3A345)

### Checklist

- [X] Added unit and Flow Tests
- [X] Added documentation + accessibility tests
- [X] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/pages/color_usage.js [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+      <MainSection
+        name="Shopping color"
+        description={`
+        Instead of the Pushpin 450, the Skycicle 500 (blue) is the primary color to indicate shopping-related features. The related token is:
+        **$color-background-shopping** - Use this token name when conveying a primary action on shopping experiences.
+        `}
+      >
+        <ColorTile description="Shopping" name="Skycicle" number={500} tokenId="blue" />
+      </MainSection>
+      {/*
```
Comment (ayeshakmaz) [inline docs/pages/color_usage.js:84]: Checking w/ Cintia on this

File: docs/components/ColorTile.js [hunk 2]
```diff
@@ -5,34 +5,42 @@ import { Box, Text } from 'gestalt';
...
 
 type Props = {|
-  name: string,
-  tokenId: string,
-  number: number,
-  textColor: 'darkGray' | 'white',
+  fullTokenName: string,
+  name?: string,
+  number?: number,
+  textColor?: 'darkGray' | 'white',
```
Comment (dangerismycat) [inline docs/components/ColorTile.js:11]: Would it be possible to simplify the API by combining `textColor` and `showName`? Something like
```js
textColor: 'darkGray' | 'white' | 'none';
```
where `'none'` is equivalent to `showName={false}`

File: docs/components/ColorTile.js [hunk 3]
```diff
@@ -5,34 +5,42 @@ import { Box, Text } from 'gestalt';
 import tokens from 'gestalt-design-tokens/dist/json/variables.json';
 
 type Props = {|
-  name: string,
-  tokenId: string,
-  number: number,
-  textColor: 'darkGray' | 'white',
+  fullTokenName: string,
```
Comment (dangerismycat) [inline docs/components/ColorTile.js:8]: What is the difference between `fullTokenName` and `name`? Should `name` be updated to `displayName`? Should `fullTokenName` just be `tokenName`?

File: docs/pages/color_usage.js [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+        />
+      </MainSection>
+      <MainSection
+        name="Secondary color"
+        description={`
+        A secondary color highlights medium to low-emphasis actions, creating a balance with the primary color. Our secondary color related token is:
+        **$color-background-secondary-base** - Use this token for secondary, medium to low-emphasis actions.
+        `}
+      >
+        <ColorTile description="Secondary base" fullTokenName="color-background-secondary-base" />
```
Comment (dangerismycat) [inline docs/pages/color_usage.js:57]: This needs dark text — I can barely read it
Screen Shot 2022-02-16 at 11 18 22 AM

File: docs/pages/color_usage.js [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+        <ColorTile
+          description="Shopping"
+          fullTokenName="color-background-shopping"
+          name="Skycicle"
+          number={500}
+        />
+      </MainSection>
+      <MainSection
+        name="Background color"
+        description="Background colors are used behind text or applied to other elements to deliver specific messages or meanings, such as information, success, warning, and error. They are divided into two categories, base colors and weak colors. "
```
Comment (dangerismycat) [inline docs/pages/color_usage.js:88]: grammar nit: "…two categories: base colors and weak colors."

File: docs/pages/color_usage.js [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+          Since they have a light tint, these colors can be used behind text.
+          **$color-background-info-weak** - Use this token as a background color to indicate neutral information.
+          **$color-background-success-weak** - Use this token as a background color to indicate success.
+          **$color-background-warning-weak** - Use this token as a background color to indicate warning alerts.
+          **$color-background-error-weak** -  Use this tokens as a background color to indicate errors.
+        `}
+        >
+          <Flex gap={4} direction="column">
+            <Flex gap={4} wrap>
+              <ColorTile description="Info weak" fullTokenName="color-background-info-weak" />
```
Comment (dangerismycat) [inline docs/pages/color_usage.js:139]: ditto here, and also these need dark text
Screen Shot 2022-02-16 at 11 21 01 AM

File: docs/pages/color_usage.js [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+      </MainSection>
+      <MainSection
+        name="Background color"
+        description="Background colors are used behind text or applied to other elements to deliver specific messages or meanings, such as information, success, warning, and error. They are divided into two categories, base colors and weak colors. "
+      >
+        <MainSection.Subsection
+          title="Base background colors"
+          description={`
+          Use on UI elements to convey a specific status or message. Usually, they aren't used behind default text.
+          **$color-background-info-base** - Use this token as a background color to indicate neutral information.
```
Comment (dangerismycat) [inline docs/pages/color_usage.js:94]: "Use the token(s)", "Use this token", etc are a bit repetitive in this section. I wonder if they're needed?

File: docs/pages/color_usage.js [hunk 8]
```diff
@@ -0,0 +1,270 @@
...
+              fullTokenName="color-text-icon-warning"
+            />
+          </Flex>
+          <Flex gap={4} wrap>
+            <ColorTile
+              description="Error"
+              textColor="white"
+              fullTokenName="color-text-icon-error"
+            />
+            <ColorTile description="Inverse" fullTokenName="color-text-icon-inverse" />
```
Comment (dangerismycat) [inline docs/pages/color_usage.js:246]: this needs dark text
Screen Shot 2022-02-16 at 11 36 33 AM

File: docs/pages/color_usage.js [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+          </Flex>
+        </Flex>
+      </MainSection>
+      <MainSection
+        name="Color and accessibility"
+        description={`
+      Please consider accessibility when using colors. When it comes to accessibility, it is our responsibility to deliver an inclusive product that inspires everyone.
```
Comment (dangerismycat) [inline docs/pages/color_usage.js:253]: "When it comes to accessibility" probably isn't necessary

File: docs/pages/color_palette.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+
+        For further Brand guidance, please reference the [Brand guidelines website](https://brand.pinterest.com/).
+        `}
+        showSourceLink={false}
+      />
+      <MainSection
+        name="Baseline colors"
+        description={`
+        Our baseline palette helps to create consistency across products.
+        The baseline palette is comprised of our hero and neutrals colors, allowing the Pinterest content to shine, while ensuring enough color contrast. In addition, these colors should be purpose-driven to provide a better user experience. Check out [Color Usage](/color_usage) for guidance.
```
Comment (dangerismycat) [inline docs/pages/color_palette.js:43]: not sure if it's a term of art, but "these colors should be purpose-driven to provide a better user experience" doesn't mean much to me?

**Review by dangerismycat — APPROVED** [review]

File: docs/components/ColorTile.js (hunk 3 above)
Comment (ayeshakmaz) [reply docs/components/ColorTile.js:8]: Simplified the API for this cmp, should be easier now

File: docs/pages/color_palette.js (hunk 10 above)
Comment (ayeshakmaz) [reply docs/pages/color_palette.js]: Updated the phrasing here

**Pull request merged**